pulsar-broker/src/test/java/org/apache/pulsar/schema/SchemaTest.java

```diff
@@ -781,4 +786,181 @@ public void testNullKey() throws Exception {
+        if (message == null) {
+            Assert.fail("Failed to receive message for topic " + topic);
+        }
+        if (!message.getReaderSchema().isPresent()) {
+            Assert.fail("Failed to get reader schema for topic " + topic);
+        }
+        message.getValue();
+
+        Schema<?> readerSchema = message.getReaderSchema().get();
+        if (readerSchema instanceof KeyValueSchema
+                && ((KeyValueSchema<?, ?>) readerSchema)
+                .getKeyValueEncodingType().equals(KeyValueEncodingType.SEPARATED)) {
+            autoProducer.newMessage(
+                    Schema.AUTO_PRODUCE_BYTES(message.getReaderSchema().get())).keyBytes(message.getKeyBytes())
+                    .value(message.getData())
+                    .properties(message.getProperties())
+                    .send();
+        } else {
+            autoProducer.newMessage(Schema.AUTO_PRODUCE_BYTES(message.getReaderSchema().get()))
+                    .properties(message.getProperties())
+                    .value(message.getData())
+                    .send();
+        }
+        producer.close();
+        consumer.close();
+    }
+
+    private void checkSchemaForAutoSchema(int index, Message<GenericRecord> message) {
+        if (!message.getReaderSchema().isPresent()) {
+            Assert.fail("Failed to get reader schema for auto consume multiple schema topic.");
+        }
+        Object nativeObject = message.getValue().getNativeObject();
+        String baseTopic = message.getProperty("baseTopic");
+        JsonNode jsonNode;
+        KeyValue<?, ?> kv;
+        switch (baseTopic) {
+            case "bytes_schema":
+                Assert.assertEquals(new String((byte[]) nativeObject), "bytes value");
+                break;
+            case "string_schema":
+                Assert.assertEquals((String) nativeObject, "string value");
+                break;
+            case "bool_schema":
+                Assert.assertEquals(nativeObject, Boolean.TRUE);
+                break;
+            case "json_one_schema":
+                jsonNode = (JsonNode) nativeObject;
+                Assert.assertEquals(jsonNode.get("id").intValue(), 1);
+                break;
+            case "json_three_schema":
+                jsonNode = (JsonNode) nativeObject;
+                Assert.assertEquals(jsonNode.get("id").intValue(), 3);
+                Assert.assertEquals(jsonNode.get("name").textValue(), "ran");
+                break;
+            case "json_four_schema":
+                jsonNode = (JsonNode) nativeObject;
+                Assert.assertEquals(jsonNode.get("id").intValue(), 4);
+                Assert.assertEquals(jsonNode.get("name").textValue(), "tang");
+                Assert.assertEquals(jsonNode.get("age").intValue(), 18);
+                break;
+            case "avro_one_schema":
+                org.apache.avro.generic.GenericRecord genericRecord =
```

Comment:
does this type is `PersonOne` not `GenericAvroRecord` is right?

Reply:
It seems that if using the `Schema.AUTO_CONSUME()`, the reader doesn't know the Pojo `PersonOne` so the decoding result is `org.apache.avro.generic.GenericRecord`.